src/Microsoft.Health.Fhir.Api/Features/Throttling/ThrottlingMiddleware.cs

```diff
@@ -55,21 +60,26 @@ public sealed class ThrottlingMiddleware : IAsyncDisposable, IDisposable
         {
             _next = EnsureArg.IsNotNull(next, nameof(next));
             _logger = EnsureArg.IsNotNull(logger, nameof(logger));
-            _configuration = EnsureArg.IsNotNull(throttlingConfiguration?.Value, nameof(throttlingConfiguration));
+            ThrottlingConfiguration configuration = EnsureArg.IsNotNull(throttlingConfiguration?.Value, nameof(throttlingConfiguration));
             EnsureArg.IsNotNull(securityConfiguration?.Value, nameof(securityConfiguration));
 
             _securityEnabled = securityConfiguration.Value.Enabled;
 
             _excludedEndpoints = new HashSet<(string method, string path)>(new StringTupleOrdinalIgnoreCaseEqualityComparer());
 
-            if (_configuration?.ExcludedEndpoints != null)
+            if (configuration?.ExcludedEndpoints != null)
             {
-                foreach (var excludedEndpoint in _configuration.ExcludedEndpoints)
+                foreach (var excludedEndpoint in configuration.ExcludedEndpoints)
                 {
                     _excludedEndpoints.Add((excludedEndpoint.Method, excludedEndpoint.Path));
                 }
             }
 
+            // snapshot the configuration values to reduce the number of instructions that need to execute in the lock.
+            _concurrentRequestLimit = configuration.ConcurrentRequestLimit;
```

Comment:
Would it be useful to validate here that `_concurrentRequestLimit` is > 0 (to avoid returning 429s for all requests)?